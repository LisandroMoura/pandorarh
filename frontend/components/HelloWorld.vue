<!-- components/HelloWorld.vue -->
<template>
  <div class="hello-world">
    <h1>{{ msg }}</h1>
    
    <p class="description">
      Componente de exemplo para testes unit√°rios no Nuxt.js
    </p>

    <button 
      @click="$emit('greet', 'Hello from Nuxt!')"
      class="greet-button"
    >
      Clique para saudar
    </button>
  </div>
</template>

<script setup>
defineProps({
  msg: {
    type: String,
    required: true
  }
});
</script>

<style scoped>
.hello-world {
  text-align: center;
  padding: 2rem;
  border: 1px solid #eaeaea;
  border-radius: 8px;
  max-width: 600px;
  margin: 0 auto;
}

h1 {
  color: #213547;
  font-size: 2em;
  margin-bottom: 1rem;
}

.description {
  color: #666;
  margin-bottom: 1.5rem;
}

.greet-button {
  background-color: #f5f5f5;
  border: 1px solid #ddd;
  padding: 0.5rem 1rem;
  border-radius: 4px;
  cursor: pointer;
  transition: background-color 0.25s;
}

.greet-button:hover {
  background-color: #e0e0e0;
}
</style>