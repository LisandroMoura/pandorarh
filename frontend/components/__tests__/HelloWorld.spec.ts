import { describe, test, expect } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import HelloWorld from '@/components/HelloWorld.vue'

describe('HelloWorld.vue', () => {
  test('renderiza a mensagem correta', () => {
    const msg = 'Bem-vindo ao Nuxt!'
    const wrapper = shallowMount(HelloWorld, {
      props: { msg }
    })
    
    expect(wrapper.text()).toContain(msg)
    expect(wrapper.html()).toMatchSnapshot()
  })
})