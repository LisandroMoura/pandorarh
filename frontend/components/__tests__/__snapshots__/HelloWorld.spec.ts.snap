// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HelloWorld.vue > renderiza a mensagem correta 1`] = `
"<div data-v-b52b555e="" class="hello-world">
  <h1 data-v-b52b555e="">Bem-vindo ao Nuxt!</h1>
  <p data-v-b52b555e="" class="description"> Componente de exemplo para testes unit√°rios no Nuxt.js </p><button data-v-b52b555e="" class="greet-button"> Clique para saudar </button>
</div>"
`;
